src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const x = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const key in obj.extraData) {
+          x[key] = obj.extraData[key];
+        }
+        result.push({ ...x });
+        break;
+
+      case 'removeProperties':
+        for (const key of obj.keysToRemove) {
+          delete x[key];
+        }
+        result.push({ ...x });
```

Comment:
[Checklist] [DONT REPEAT YOURSELF] - If you perform same action in all `switch` cases - do it just once afterwards.

Reply:
in that case two tests don't work. left cycle be)